proposed/0007-Address_relocation.md

```diff
@@ -0,0 +1,40 @@
+- Feature Name: Prevent any single address being used more than once during address relocation.
+- Type enhancement
+- Related components routing
+- Start Date: 21-09-2015
+- RFC PR: (leave this empty)
+- Issue number: (leave this empty)
+
+# Summary
+
+Prevent a potential attack where a node gains overall/majority influence in a group setting.
+
+# Motivation
+
+The possibility 'may' exist for a node on the network to generate off-line derived identities close to a specific target and to then connect nodes whose resultant network relocated identities actually lie close to that target. In such cases the owner of that node will dominate, or at least have a dominant influence within, the group close to that target. This RFC proposes an amendment to the current process to prevent this possibility arising in practice.
+
+# Detailed design
+
+A node joining the SAFE network generates a new cryptographic key pair to access the network for each session. The generated public key, K, is then cryptographically hashed using some hash function H to give H(K). The node then sends a message, containing K, to the group H(K) requesting a network identifier for the session. On receipt of such a request each node belonging to the group hashes the concatenation of K with the two closest nodes, N1, N2 say, in xor distance from K present in their routing tables, giving H(K + N1 + N2) = N, the identifier the node will use on the network for the session.
```

Comment:
The complexity of attack under `K closer to T` and `Hash(K + N1 + N2) closer to T` is the same